.gitattributes

```diff
@@ -1,3 +1,14 @@
-/examples export-ignore
-/docs export-ignore
+* text=auto
```

Comment:
I don't think we should add this since it changes the behavior of how files are committed in the repository regardless of the operating system you're working on. According to the documentation:

>Setting the text attribute on a path enables end-of-line normalization and marks the path as a text file. End-of-line conversion takes place without guessing the content type.
>
>If the text attribute is unspecified, Git uses the core.autocrlf configuration variable to determine if the file should be converted.

If I don't recall wrong we have some files that must be treated as binary, and some others whose line endings should not change since we're relying on them being LF in the tests. Indeed normalization of things is always a good thing imho, but in this case I believe we should not force this setting to everyone. If line endings change for some reason it will be reported anyway in the diff

Reply:
https://help.github.com/articles/dealing-with-line-endings/#per-repository-settings
`* text=auto` = `Git will handle the files in whatever way it thinks is best. This is a good default option.`
it's not affect binary or unknown file types, but prevent bad configuration of `core.autocrlf`

Reply:
If it doesn't affect binaries, we should be good.

Reply:
it should not impact binary or unknown files

Reply:
After having a look at how other repositories use this feature and at the Git documentation please add the following content too, then LGTM!

```gitattributes
*.php text whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4 diff=php
*.json text whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4
*.yml text whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4
*.md text whitespace=blank-at-eol,blank-at-eof
*.rst text whitespace=blank-at-eol,blank-at-eof
```

Reply:
i got inspiration from https://github.com/composer/composer/blob/master/.gitattributes and add `eol=lf` to force always to use `LF` for this files